docker_auth.go

```diff
@@ -42,6 +43,11 @@ func dockerImageAuth(ctx context.Context, image string, configs map[string]regis
 	defaultRegistry := defaultRegistryFn(ctx)
 	reg := core.ExtractRegistry(image, defaultRegistry)
 
+	// Make comparison case-insensitive and handle both aliases
+	if strings.EqualFold(reg, "docker.io") || strings.EqualFold(reg, "registry.hub.docker.com") {
+		reg = defaultRegistry
+	}
+
```

Comment:
question: should this be handled by `core.ExtractRegistry`?

Reply:
It could. Up to y'all